pilot/pkg/xds/workload.go

```diff
@@ -0,0 +1,248 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"strconv"
+
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+
+	"istio.io/api/security/v1beta1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/security/authn/factory"
+	"istio.io/istio/pilot/pkg/util/protoconv"
+	"istio.io/istio/pkg/config/schema/kind"
+	workloadapi "istio.io/istio/pkg/workloadapi"
+)
+
+type WorkloadGenerator struct {
+	s *DiscoveryServer
+}
+
+var (
+	_ model.XdsResourceGenerator      = &WorkloadGenerator{}
+	_ model.XdsDeltaResourceGenerator = &WorkloadGenerator{}
+)
+
+func (e WorkloadGenerator) GenerateDeltas(
```

Comment:
I don't see calls to `AmbientWorkloads` anywhere. Are the changes to workloadcache relevant? I'm not opposed to doing away with it and converging on a single "kube controller".